libs/route-config/src/lib/route-data/route-data.directive.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  Directive,
+  EmbeddedViewRef,
+  Input,
+  OnDestroy,
+  OnInit,
+  TemplateRef,
+  ViewContainerRef,
+} from '@angular/core';
+import { distinctUntilChanged, map, switchMap, takeUntil, tap } from 'rxjs/operators';
+import { BehaviorSubject, Subject } from 'rxjs';
+import { RouteConfigService, RouteData } from '../route-config.service';
+
+export interface RouteDataDirectiveContext<C> {
+  $implicit: C;
+}
+
+@Directive({
+  selector: '[tdRouteData]',
+})
+export class RouteDataDirective<C extends RouteData> implements OnInit, OnDestroy {
+  private destroy$ = new Subject();
+
+  private defaultValue$ = new BehaviorSubject<Partial<C> | null>(null);
+  private view!: EmbeddedViewRef<RouteDataDirectiveContext<Partial<C>>>;
+
+  private data$ = this.defaultValue$.pipe(
+    map((defaultValue) => defaultValue || {}),
```

Comment:
why don't you just init the BehaviorSubject with `{}` ?

Reply:
done